class.c

```diff
@@ -1059,6 +1059,29 @@ Perl_class_add_ADJUST(pTHX_ HV *stash, CV *cv)
     av_push(aux->xhv_class_adjust_blocks, (SV *)cv);
 }
 
+OP *
+Perl_ck_classname(pTHX_ OP *o)
+{
+    if(!CvIsMETHOD(PL_compcv))
+        croak("Cannot use __CLASS__ outside of a method or field initializer expression");
```

Comment:
Pardon my ignorance 😃 I think `ADJUST` is currently considered a method, but eventually will become a phaser. It is intended that `__CLASS__` will work there?

Reply:
Yeah definitely. The ADJUST blocks will become individual blocks that are all still part of one overall containing method, so `__CLASS__` still still work fine there. It'll just stop being one method *per block*.